java-sdk/radar-schemas-registration/src/main/java/org/radarbase/schema/registration/KafkaTopics.java

```diff
@@ -193,40 +204,42 @@ public boolean createTopics(Stream<String> topicsToCreate, int partitions, short
     public boolean refreshTopics() throws InterruptedException {
         ensureInitialized();
         logger.info("Waiting for topics to become available.");
-        int sleep = 10;
+        int sleep = 2;
+        long timeout = 5;
         int numTries = 10;
 
         topics = null;
-        ListTopicsOptions opts = new ListTopicsOptions().listInternal(true);
+        ListTopicsOptions opts = new ListTopicsOptions()
+                .listInternal(true);
+
         for (int tries = 0; tries < numTries; tries++) {
+            long start = System.currentTimeMillis();
             try {
-                topics = getKafkaClient().listTopics(opts).names().get(sleep, TimeUnit.SECONDS);
-            } catch (ExecutionException e) {
-                logger.error("Failed to list topics from brokers: {}."
-                        + " Trying again after {} seconds.", e, sleep);
-                Thread.sleep(sleep * 1000L);
-                sleep = Math.min(MAX_SLEEP, sleep * 2);
-                continue;
-            } catch (TimeoutException e) {
-                // do nothing
+                topics = getKafkaClient()
+                        .listTopics(opts)
+                        .names()
+                        .get(sleep, TimeUnit.SECONDS);
+            } catch (ExecutionException ex) {
+                logger.error("Failed to list topics from brokers: {}", ex.getCause().toString());
+            } catch (TimeoutException ex) {
+                logger.error("Failed to list topics within {} seconds", timeout);
```

Comment:
can this be sleep? Since that is the time specified in the get call. Otherwise the get call should also use timeout?

Reply:
Indeed, done.